packages/api/src/command/medical/document/process-doc-query-webhook.ts

```diff
@@ -0,0 +1,103 @@
+import { errorToString } from "@metriport/core/util/error/index";
+import { capture } from "@metriport/core/util/notifications";
+import { out } from "@metriport/core/util/log";
+import { DocumentQueryProgress } from "../../../domain/medical/document-query";
+import { Patient } from "../../../domain/medical/patient";
+import { processPatientDocumentRequest } from "./document-webhook";
+import { MAPIWebhookStatus } from "./document-webhook";
+import { getAllDocRefMappingByRequestId } from "../docref-mapping/get-docref-mapping";
+import { getDocuments } from "../../../external/fhir/document/get-documents";
+import { toDTO, DocumentReferenceDTO } from "../../../routes/medical/dtos/documentDTO";
+import { getAllWebhookRequestByRequestId } from "../../webhook/webhook-request";
+import { Config } from "../../../shared/config";
+
+const { log } = out(`Doc Query Webhook`);
+const isSandbox = Config.isSandbox();
+
+/**
+ * Processes the document query progress to determine if when to send the document download and conversion webhooks
+ */
+export const processDocQueryProgressWebhook = async ({
+  patient,
+  documentQueryProgress,
+  requestId,
+}: {
+  patient: Pick<Patient, "id" | "cxId" | "externalId">;
+  documentQueryProgress: DocumentQueryProgress;
+  requestId: string;
+}): Promise<void> => {
+  const { id: patientId } = patient;
+
+  try {
+    const downloadStatus = documentQueryProgress.download?.status;
+    const convertStatus = documentQueryProgress.convert?.status;
+
+    const webhooks = await getAllWebhookRequestByRequestId(requestId);
+    const downloadWebhookSent = webhooks.some(
+      webhook => webhook.type === "medical.document-download"
+    );
+    const convertWebhookSent = webhooks.some(
+      webhook => webhook.type === "medical.document-conversion"
+    );
+
+    if (!downloadWebhookSent && !isSandbox) {
```

Comment:
So we do send a webhook in sandbox just not with the new flow. I left the existing flow we currently have because we dont have a docRef mapping in sandbox

Reply:
I'm confused with your answer - it says we send a WH w/ the new flow and then says we're using the existing flow. And when I look at the file history, there's a lot of entries including adding/removing `console.log`, I don't know what's going on there. 😅

Reply:
In prod we send webhooks with the new flow where we query the db for doc ref mappings, find those in FHIR and then send to the cx. 

In sandbox mode specifically `document-query-sandbox.ts` im leaving it as is. Meaning it appends progress but the WH request will not be sent from `appendDocQueryProgress` but rather `processPatientDocumentRequest` will be called in this function.